include/oneapi/dpl/internal/dynamic_selection_traits.h

```diff
@@ -299,6 +311,14 @@ struct report_value
 template <typename S, typename Info>
 inline constexpr bool report_value_v = report_value<S, Info>::value;
 
+template <typename S>
+struct lazy_report_value
```

Comment:
I believe this is the first trait for a backend -- i.e. that a backend requires a call to lazy report. Backends are not as visible as policies so likely don't want a trait in this header. End users cannot access the backend, only the policy implementation can.

Reply:
Not sure how to check if the backend has certain functions without traits. Created a new namespace called backend_traits for checking presence of lazy_report and has_enable_profiling in the backend from the policy